crates/electrum/src/electrum_ext.rs

```diff
@@ -165,20 +165,15 @@ pub trait ElectrumExt<A> {
     /// The scan for each keychain stops after a gap of `stop_gap` script pubkeys with no associated
     /// transactions. `batch_size` specifies the max number of script pubkeys to request for in a
     /// single batch request.
-    fn scan<K: Ord + Clone>(
+    fn sync_with_keychain<K: Ord + Clone>(
         &self,
-        prev_tip: Option<CheckPoint>,
         keychain_spks: BTreeMap<K, impl IntoIterator<Item = (u32, ScriptBuf)>>,
-        txids: impl IntoIterator<Item = Txid>,
-        outpoints: impl IntoIterator<Item = OutPoint>,
-        stop_gap: usize,
-        batch_size: usize,
     ) -> Result<ElectrumUpdate<K, A>, Error>;
 
     /// Convenience method to call [`scan`] without requiring a keychain.
     ///
     /// [`scan`]: ElectrumExt::scan
-    fn scan_without_keychain(
+    fn sync(
```

Comment:
Here :
```
fn sync(
        &self,
        prev_tip: Option<CheckPoint>,
        misc_spks: impl IntoIterator<Item = ScriptBuf>,
        txids: impl IntoIterator<Item = Txid>,
        outpoints: impl IntoIterator<Item = OutPoint>,
        batch_size: usize,
    )
    ```
    
    The method arguments `prev_tip, txids, outpoints, batch_size` are not used. Should I remove them also?

Reply:
@LLFourn @vladimirfomene

Reply:
WIth sync you definitely need `prev_tip` and `batch_size` but I don't think you need `stop_gap` since you are meant to exhaustively iterate through `misc_spks`.